src/Components/Replication/ReplicationSettingsForm.tsx

```diff
@@ -411,94 +506,123 @@ export default class ReplicationSettingsFormReplication extends React.Component<
           <Form.Item
             label={
               <span>
-                Filter Records&nbsp;
-                <Tooltip title="Set the labels a record should include/exclude to be replicated. If empty, all records are replicated.">
+                When Condition&nbsp;
+                <Tooltip title="JSON condition for filtering records based on labels. Example: {'&temperature': {'$gt': 25}, '&humidity': {'$lt': 60}}. Use & for labels and $ for operators. Supported operators: $gt, $gte, $lt, $lte, $eq, $ne, $and, $or">
                   <InfoCircleOutlined />
                 </Tooltip>
               </span>
             }
+            name="when"
             className="ReplicationField"
+            style={{ marginBottom: "24px" }}
           >
-            <Form.List name="recordSettings">
-              {(fields, { add, remove }) => (
-                <>
-                  {fields.map((field, index) => (
-                    <Form.Item key={`recordSettings-${field.key}-${index}`}>
-                      <div
-                        style={{
-                          display: "flex",
-                          justifyContent: "space-between",
-                          alignItems: "center",
-                          marginBottom: "10px",
-                        }}
-                      >
+            <Input.TextArea
+              rows={5}
+              placeholder="Enter JSON condition (e.g., {'&temperature': {'$gt': 25}, '&humidity': {'$lt': 60}})"
+              disabled={readOnly}
+              style={{ resize: "none" }}
+            />
+          </Form.Item>
+
+          {/* Show Filter Records section only for existing tasks */}
```

Comment:
Show Filter Records section only for existing tasks  AND  if they are already set